src/features/proposers/components/AddProposer.tsx

```diff
@@ -0,0 +1,194 @@
+  const [isLoading, setIsLoading] = useState<boolean>(false)
+  const [addDelegate] = useAddDelegateMutation()
+  const dispatch = useAppDispatch()
+
+  const chainId = useChainId()
+  const wallet = useWallet()
+  const safeAddress = useSafeAddress()
+
+  const methods = useForm<DelegateEntry>({
+    defaultValues: {
+      [DelegateEntryFields.address]: '',
+      [DelegateEntryFields.name]: '',
+    },
+    mode: 'onChange',
+  })
+
+  const { handleSubmit } = methods
+
+  const onConfirm = handleSubmit(async (data: DelegateEntry) => {
+    if (!wallet) return
+
+    setError(undefined)
+    setIsLoading(true)
+
+    try {
+      const signer = await getAssertedChainSigner(wallet.provider)
+      const typedData = getDelegateTypedData(chainId, data.address)
+      const signature = await signTypedData(signer, typedData)
+
+      await addDelegate({
+        chainId,
+        delegator: wallet.address,
+        signature,
+        label: data.name,
+        delegate: data.address,
+        safeAddress,
+      })
+
+      trackEvent(SETTINGS_EVENTS.PROPOSERS.SUBMIT_ADD_PROPOSER)
+
+      dispatch(
+        showNotification({
+          variant: 'success',
+          groupKey: 'add-proposer-success',
+          title: 'Proposer added successfully!',
+          message: `${shortenAddress(data.address)} can now suggest transactions for this account.`,
+        }),
+      )
+    } catch (error) {
+      setIsLoading(false)
+      setError(error as Error)
+      return
+    }
+
+    setIsLoading(false)
+    onSuccess()
+  })
+
+  const onSubmit = (e: BaseSyntheticEvent) => {
+    e.stopPropagation()
+    onConfirm(e)
+  }
+
+  const onCancel = () => {
+    trackEvent(SETTINGS_EVENTS.PROPOSERS.CANCEL_ADD_PROPOSER)
+    onClose()
+  }
+
+  return (
+    <Dialog open onClose={onCancel}>
+      <FormProvider {...methods}>
+        <form onSubmit={onSubmit}>
+          <DialogTitle>
+            <Box data-testid="untrusted-token-warning" display="flex" alignItems="center">
+              <Typography variant="h6" fontWeight={700} sx={{ display: 'flex', alignItems: 'center', gap: 1 }}>
+                Add proposer
+              </Typography>
+
+              <Box flexGrow={1} />
+
+              <IconButton aria-label="close" onClick={onCancel} sx={{ marginLeft: 'auto' }}>
+                <Close />
+              </IconButton>
+            </Box>
+          </DialogTitle>
+
+          <Divider />
+
+          <DialogContent>
+            <Box mb={2}>
+              <Typography variant="body2">
+                You&apos;re about to grant this address the ability to propose transactions. To complete the setup,
+                confirm with a signature from your connected wallet.
+              </Typography>
+            </Box>
+
+            <Alert severity="info">Proposer’s name and address will be publicly visible.</Alert>
+
+            <Box my={2}>
+              <AddressInput name="address" label="Address" variant="outlined" fullWidth required />
```

Comment:
I was surprised that it is possible to add the Safe or even Owners as proposers. It does "break" our frontend logic in that if I add an owner as a proposer our UI will tell me that I am only proposing even though the transaction is signed at the end. Lets add validation to prevent adding owners and the safe itself as proposers.